pallets/runtime/tests/src/fee_details.rs

```diff
@@ -78,25 +85,111 @@ fn cdd_checks() {
             assert_err!(
                 CddHandler::get_valid_payer(
                     &Call::MultiSig(multisig::Call::accept_multisig_signer_as_key(alice_auth_id)),
-                    &alice_account_signatory
+                    &alice_account
                 ),
                 InvalidTransaction::Custom(TransactionError::CddRequired as u8)
             );
 
+            // call to remove authorisation with issuer paying should fail if issuer does not have a valid cdd
+            assert_err!(
+                CddHandler::get_valid_payer(
+                    &Call::Identity(identity::Call::remove_authorization(
+                        alice_account_signatory.clone(),
+                        alice_auth_id,
+                        true
+                    )),
+                    &alice_account
+                ),
+                InvalidTransaction::Custom(TransactionError::CddRequired as u8)
+            );
+
+            // call to remove authorisation with caller paying should fail if caller does not have a valid cdd
+            assert_err!(
+                CddHandler::get_valid_payer(
+                    &Call::Identity(identity::Call::remove_authorization(
+                        alice_account_signatory.clone(),
+                        alice_auth_id,
+                        false
+                    )),
+                    &alice_account
+                ),
+                InvalidTransaction::Custom(TransactionError::CddRequired as u8)
+            );
+
+            // check that authorisation can be removed correctly
+            let _musig_address2 =
+                MultiSig::get_next_multisig_address(AccountKeyring::Charlie.public());
+            assert_ok!(MultiSig::create_multisig(
+                charlie_signed.clone(),
+                vec![Signatory::Account(AccountKeyring::Alice.public())],
+                1,
+            ));
+            let alice_auth_id = get_last_auth_id(&alice_key_signatory);
+
+            // call to remove authorisation with caller paying should fail if caller does not have a valid cdd
+            assert_err!(
+                CddHandler::get_valid_payer(
+                    &Call::Identity(identity::Call::remove_authorization(
+                        alice_account_signatory.clone(),
+                        alice_auth_id,
+                        false
+                    )),
+                    &alice_account
+                ),
+                InvalidTransaction::Custom(TransactionError::CddRequired as u8)
+            );
+
+            // call to remove authorisation with issuer paying should succeed as issuer has CDD
+            assert_eq!(
+                CddHandler::get_valid_payer(
+                    &Call::Identity(identity::Call::remove_authorization(
+                        alice_account_signatory,
+                        alice_auth_id,
+                        true
+                    )),
+                    &alice_account
+                ),
```

Comment:
Maybe extract this part into another closure too, returning `DispatchError`.

Reply:
ditto